python/src/opendp/_extrinsics/_make_np_sum/make_np_sum.tex

```diff
@@ -0,0 +1,90 @@
+
+
+\documentclass{article}
+\input{../../../../../rust/src/lib.sty}
+
+
+
+\title{\texttt{fn make\_np\_sum}}
+\author{Michael Shoemate}
+\begin{document}
+
+\maketitle
+
+\contrib
+
+Proves soundness of \texttt{make\_np\_sum} in \asOfCommit{__init__.py}{f5bb719}.
+
+\texttt{make\_np\_sum} accepts an \texttt{input\_domain} and \texttt{input\_metric},
+and returns a stable Transformation that computes the vector-valued sum with bounded $L_p$ sensitivity.
+
+\subsection*{PR History}
+\begin{itemize}
+    \item \vettingPR{490}
+\end{itemize}
+
+\section{Hoare Triple}
+
+\subsection*{Preconditions}
+None
+
+\subsection*{Pseudocode}
+\lstinputlisting[language=Python,escapechar=|,firstline=9,lastline=49]{./__init__.py}
+
+\subsection*{Postcondition}
+
+\validTransformation{\texttt{(input\_domain, input\_metric)}}{\\ \texttt{make\_np\_sum}}
+
+\section{Proof}
+
+\begin{proof} 
+
+Let $x$ and $x'$ be datasets that are \texttt{d\_in}-close with respect to \texttt{input\_metric}.
+By \ref{domain-check}, the input domain is \texttt{NPArray2Domain},
+and by \ref{metric-check} input metric is the \texttt{SymmetricDistance}. 
+By \ref{norm-check} and \ref{p-check}, the input data has row-p-norm bounded by at most \texttt{norm}, which we'll refer to as $R$,
+centered at \texttt{origin}, which we'll refer to as $O$.
+
+\subsection{Appropriate Output Domain}
+
+Since the input data is a 2-dimensional numpy array, then the output is a 1-dimensional numpy array of the same type.
+Therefore the output is a member of the output vector domain.
+
+\subsection{Stability Guarantee}
+
+Now consider two cases: when the data set size is known, and when it is not known.
+
+\subsubsection{Known Size}
+    WLOG, assume $x, x'$ only differ on the leading $\lfloor d_{in} / 2 \rfloor$ elements.
+
```

Comment:
Why is it WLOG? Why is it din/2 when din elements are contributed by the user? Might want to clarify.

Reply:
I left more comments in the proof.